systems/plants/setupLCPmex.cpp

```diff
@@ -169,14 +169,14 @@ bool callFastQP(MatrixBase<DerivedM> const & M, MatrixBase<Derivedw> const & w,
     return false;
   }
 
-  // getThresholdInclusion(Ain * zqp - bin, -SMALL, ineq_violations);
-  // getThresholdInclusion(beq - Aeq.transpose() * zqp, -SMALL, violations);
-
-  // for (size_t i = 0; i < num_inactive_z; i++) { 
-  //   if (ineq_violations[i] && violations[i]) { 
-  //     return false;
-  //   }
-  // }
+  getThresholdInclusion(Ain * zqp - bin, -SMALL, ineq_violations);
+  getThresholdInclusion(beq - Aeq.transpose() * zqp, -SMALL, violations);
```

Comment:
I think you want to evaluate the matrix expressions before passing them into getThresholdInclusion. I think what happens now is that getThresholdInclusion is instantiated with some weird Eigen expression type, and when you ask for values[x] inside the loop on line 32, the whole thing gets evaluated on every iteration.

Also, it's a good idea to separate the operations instead of performing everything at once, e.g. as
```
VectorXd bla = Ain * zqp;
bla -= bin
```
so as to avoid temporaries, according to http://eigen.tuxfamily.org/dox/TopicWritingEfficientProductExpression.html